tests/kafkatest/services/kafka/templates/kafka.properties

```diff
@@ -52,6 +52,7 @@ quota.consumer.bytes.per.second.overrides={{ quota_config.quota_consumer_bytes_p
 
 security.inter.broker.protocol={{ interbroker_security_protocol }}
 
+ssl.client.auth=required
```

Comment:
I chatted with @fpj about this before the break as I was surprised it worked too. It does as we don't check it unless security is enabled. However it might be clearer to push it into the authoriser conditional.